server/server.go

```diff
@@ -946,3 +948,80 @@ func isNotFound(err error) bool {
 
 	return false
 }
+
+// startWatcherForMirrorRegistries sets up a file watcher to monitor changes
+// in the "registries.conf.d" directory (default: "/etc/containers/registries.conf.d").
+// It then delegates the monitoring task to watchAndReloadMirrorRegistriesConfiguration,
+func (s *Server) startWatcherForMirrorRegistries(ctx context.Context, registriesConfDDir string) {
+	if registriesConfDDir == "" {
+		log.Infof(ctx, "No registries.conf.d directory specified, defaulting to /etc/containers/registries.conf.d")
+		registriesConfDDir = "/etc/containers/registries.conf.d"
+	}
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		log.Fatalf(ctx, "Failed to create new watcher: %v", err)
+	}
+	defer watcher.Close()
+
+	log.Infof(ctx, "Registered reload watcher for mirror registries configuration")
+
+	if err := watcher.Add(registriesConfDDir); err != nil {
+		log.Errorf(ctx, "Failed to add watcher for path %q: %s", registriesConfDDir, err)
+		return
+	}
+
+	s.watchAndReloadMirrorRegistriesConfiguration(ctx, watcher)
+}
+
+func (s *Server) watchAndReloadMirrorRegistriesConfiguration(ctx context.Context, watcher *fsnotify.Watcher) {
+	var timer *time.Timer
+	reloadChannel := make(chan string, 1)
+	reloadChannel <- ""
+
+	go func() {
+		// The for loop ensures that the channel is properly drained, even if
+		// no new events are received, thus preventing potential deadlocks.
+		// For each event name received, the goroutine checks if it's not
+		// an empty string and then reloads the registries.
+		for evenName := range reloadChannel {
+			if evenName != "" {
```

Comment:
Aside from the initialisation (a few lines above), when did you see an empty event? I wonder if there is a race with the new (a few lines below) goroutine? Not sure.

Reply:
I removed that logic altogether.